llama_stack/providers/utils/inference/model_registry.py

```diff
@@ -82,9 +82,37 @@ def get_provider_model_id(self, identifier: str) -> str | None:
     def get_llama_model(self, provider_model_id: str) -> str | None:
         return self.provider_id_to_llama_model_map.get(provider_model_id, None)
 
+    async def query_available_models(self) -> list[str]:
```

Comment:
Maybe something like check_model_availability(model) -> bool would also suffice instead of fetching all models?